.azure/gpu-tests.yml

```diff
@@ -88,10 +92,20 @@ jobs:
       displayName: 'Install dependencies'
       condition: eq(variables['continue'], '1')
 
+    # TODO
```

Comment:
Instead of completely skipping Bagua installation, we can skip the problematic tests if we are in the newer GPU architecture

Reply:
Good point, so do we have some safe tests if the tests would run... like try/catch, so it ll be a rather general solution regardless of the machine you run it on?

Reply:
Try catch won't work since segfaults just mess everything up.

I suggest skipping the tests instead

```python
    if torch.cuda.get_device_capability()[0] >= 8:
        pytest.skip("Bagua does not support this architecture")
```

Reply:
I see; I am rather thinking instead of hardcode test just for one CUDA capacity to have in the CI flow check that the installed version is compatible given HW, which would yield in something like env variable, and this will be parsed in tests

```bash
BAGUA_GPU_COMPATIBLE=$(python -c "print(int(torch.cuda.get_device_capability()[0] < 8))")
```
and in code
```py
if not bool(os.environ.get("BAGUA_GPU_COMPATIBLE", 1)):
        pytest.skip("Bagua does not support this architecture")
```

Reply:
The Bagua test file can define this condition and all tests use it. Does t need to be an environment variable

Reply:
Well, my point is the separate environment from code... so the model situation is that in a few weeks the compile Bagua with this GPU capacity, and we need to edit tests, and I would rather update the CI pipeline... not sure if I describe it well :(

Reply:
wouldn't it then work to simply skip installing bagua in those envs (as it wouldn't be valid anyways)? The tests should only be run if bagua is installed

Reply:
@justusschock AFAIK not all Bagua algorithms fail in the new architecture. Not installing Bagua means the strategy is now completely untested

> and I would rather update the CI pipeline

What's the issue with updating a test attribute? I find it confusing defining it as an environment variable. It's a leaky way to do it.

Reply:
I see; I am just thinking if there was runtime to determine if the used HW is compatible with the installed Bagua version...
As you said, the compatibility raises segmentation error, see, we could have a trivial script to run in a separate CI step, and if it passes, we will run all Bagua, and if it fails, we skip all...
But you are right that atm would mean no Bagua will be tested
Also not sure if the CI step failed would propagate as complete workflow fail :(

Reply:
@carmocca I get that, but if it is segfaulting due to the gpu architecture, I wouldn't trust the results with other algorithms as well and might expect this to be flaky at some point.

Reply:
If we gets worse, we can always choose to leave it untested. But having some tests at least checks that the strategy runs at all.